### ISSUE TITLE
FixedBudgetInteractionLearner's unweighted top-k utility lets inactive-task zero placeholders outrank real active signals

### ISSUE THREAD

**Issue by Svector-anu** [SlopDotCash/asi#275]
## What's wrong

\`FixedBudgetInteractionLearner\`'s unweighted \`topk\` utility reducers zero out inactive task rows (\`weighted_activity = weighted_activity * task_weights[:, None]\`, where \`task_weights\` is 0 for inactive tasks) before selecting the top-k values via \`jnp.sort(...)[-k:, :]\`.

For the **signed** feature-signal reducer, this is a real correctness bug: a synthetic zero placeholder for an inactive task can outrank a legitimate *negative* measurement from an active task in the naive ascending-sort top-k selection, hiding the real (harmful) signal entirely.

Reproduced directly before any fix:

\`\`\`text
active=[True, False, False], utility_top_k=2
signed_reductions=[[-4.0], [99.0], [99.0]]  # tasks 2,3 inactive, zeroed to 0
reported [0.]
expected [-4.]
\`\`\`

With active task 1 reporting \`-4\` and inactive tasks 2/3 zeroed to \`0\`, sorted ascending is \`[-4, 0, 0]\`, top-2 selects \`[0, 0]\`, mean = \`0\` — the genuine \`-4\` signal is silently dropped.

The same eligibility gap also **dilutes** nonnegative current-utility values toward zero when fewer than \`k\` tasks are active: with only 1 active task producing a real value of \`6.0\` and \`top_k=2\`, the old code averaged in a zero-padded inactive slot, returning \`3.0\` instead of the correct \`6.0\`.

## Reachability

\`FixedBudgetInteractionLearner\` is genuinely package-root exported (\`alberta_framework/__init__.py\`, both import and \`__all__\`). The affected signed reducer is called from the public \`.update\` path after \`one_step_output_loss_reduction\` computes per-task feature measurements. It's also constructed by \`alberta_framework/recurring_feature_gate.py\` and \`alberta_framework/core/prototype_feature_lifecycle.py\`. Real non-hardcoded-safe inputs reach the bug whenever active-task masking leaves fewer than \`utility_top_k\` valid targets and an active task's signed loss reduction is negative.

## Fix

Inactive tasks are now masked to \`-inf\` (not zero) before top-k selection, guaranteeing an inactive placeholder can never outrank a genuine active value regardless of sign. The mean now divides by the actual count of genuinely-selected finite (active) entries, not blindly by \`k\`. The historical \`utility_task_balancing="none"\` branch is left byte-for-byte algebraically unchanged.

PR incoming with a regression test proving a single active task's real \`-4.0\` signal survives \`top_k=2\` selection against four inactive placeholders, plus a mutation-resistance transcript.
